src/js/Content/Features/Community/Inventory/FInventoryMarketHelper.js

```diff
@@ -152,70 +152,88 @@ export default class FInventoryMarketHelper extends Feature {
         }
     }
 
-    _setBackgroundOption(thisItem, assetId, itemActions) {
+    async _setBackgroundOption(thisItem, itemActions) {
 
-        if (!document.querySelector(".inventory_links")) { return; }
         if (itemActions.querySelector(".es_set_background")) { return; }
 
+        // Make sure the selected item is a background
         const viewFullBtn = itemActions.querySelector("a");
-        if (!viewFullBtn) { return; }
+        if (!viewFullBtn || !/public\/images\/items/.test(viewFullBtn.href)) { return; }
 
-        if (!/public\/images\/items/.test(viewFullBtn.href)) { return; }
+        // Get owned backgrounds and the communityitemid for equipped background
+        if (!this.profileBgsOwned) {
+
+            // Retrieve access token from user's profile to use in Steam API calls
+            const html = await RequestData.getHttp(User.profileUrl);
```

Comment:
Can't this be included in `Promise.all`?

Reply:
Can it? The user token is required to make those API calls, they can't be done at the same time.
Also what do you think of moving this to background, or the User class, so it can be called like `User.getUserToken()`? See https://github.com/tfedor/AugmentedSteam/issues/1021#issuecomment-769078402.